lib/Dialect/FIRRTL/OpFolds.cpp

```diff
@@ -50,6 +50,27 @@ OpFoldResult ConstantOp::fold(ArrayRef<Attribute> operands) {
   return valueAttr();
 }
 
+OpFoldResult DivPrimOp::fold(ArrayRef<Attribute> operands) {
+  APInt value;
+
+  /// div(x, x) -> 1
+  ///
+  /// Division by zero is undefined in the FIRRTL specification. This
+  /// fold exploits that fact to optimize self division to one.
+  if (lhs() == rhs()) {
+    auto width = getType().cast<IntType>().getWidthOrSentinel();
+    auto type = IntegerType::get(getContext(), width);
```

Comment:
This can be simplified to `auto type = getType();`

Reply:
and then inlined into the next line if it "looks good" that way.

Reply:
I'm having some issues with `auto type = getType()`. That type needs to be an int or a float and it's not. I'm sure there's some way to coerce it that's terser than using `IntegerType::get`. I'll simplify it for now by inlining `type`.